components/pd_client/src/util.rs

```diff
@@ -456,13 +493,40 @@ pub async fn try_connect_leader(
         }
     }
 
-    // Then try to connect the PD cluster leader.
     if let Some(resp) = resp {
+        // Then try to connect the PD cluster leader.
         let leader = resp.get_leader().clone();
         for ep in leader.get_client_urls() {
             if let Ok((client, _)) = connect(Arc::clone(&env), &security_mgr, ep.as_str()).await {
                 info!("connected to PD leader"; "endpoints" => ep);
-                return Ok((client, resp));
+                return Ok((client, resp, ep.to_string()));
+            }
+        }
+
+        // Otherwise try to connect the PD cluster follower.
```

Comment:
How about retrying some times before falling back to follower?

Reply:
make sense